elfo-configurer/src/lib.rs

```diff
@@ -117,33 +172,55 @@ impl Configurer {
         };
 
         let config = match config {
-            Ok(config) => config,
-            Err(error) => {
-                error!(%error, "invalid config");
-                return Err(vec![ReloadConfigsError { reason: error }]);
-            }
-        };
-
-        let config = match Deserialize::deserialize(config) {
             Ok(config) => config,
             Err(error) => {
                 error!(%error, "invalid config");
                 return Err(vec![ReloadConfigsError {
-                    reason: error.to_string(),
+                    group: None,
+                    reason: error,
                 }]);
             }
         };
 
+        Deserialize::deserialize(config).map_err(|error| {
+            error!(%error, "invalid config");
+            vec![ReloadConfigsError {
+                group: None,
+                reason: error.to_string(),
+            }]
+        })
+    }
+
+    async fn load_and_check_configs(&self) -> Result<(), Vec<ReloadConfigsError>> {
+        let configs = self.load_configs().await?;
+
+        let system_configs = match_configs(&self.topology, &configs, TopologyFilter::System);
+        self.validate_all(&system_configs).await?;
```

Comment:
Maybe check against both system and user groups to collect all possible errors at once (`TopologyFilter::All`)?